rust/src/action.rs

```diff
@@ -856,6 +855,238 @@ pub enum OutputMode {
     Update,
 }
 
+/// JobInfo contained in the CommitInfo
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct JobInfo {
+    /// The job identifier
+    pub job_id: String,
+    /// The job name
+    pub job_name: String,
+    /// The run identfiier
+    pub run_id: String,
+    /// The job owner identifier
+    pub job_owner_id: String,
+    /// The trigger type
+    pub trigger_type: String,
+}
+
+/// NoteBookInfo contained in the CommitInfo
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct NotebookInfo {
+    /// The notebook identifier
+    pub notebook_id: String,
+}
+
+/// CommitInfo holds provenance information about changes to the table. This action
+/// is not stored in the checkpoint and has reduced compatibility guarantees.
+/// Information stored in it is best effort.
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct CommitInfo {
```

Comment:
Here is the quote from the spec:

> Implementations are free to store any valid JSON-formatted data

Looks like we should be modeling the content as `serde_json::Map` instead? The fields we have right now are databricks specific. An alternative implementation could write arbitrary fields, so if we model it as a static Rust struct, we won't be able to capture those.

Reply:
Do you think I need to replace the `CommitInfo`  structure with a  `serde_json::Map` which will contain the `CommitInfo` JSON attribute as it is inside the transaction log? This will make it easier to get it inside the Python binding using a simple dictionary. On the other hand, the content of the `CommitInfo` will not be checked.

Reply:
Yes, I think so. I think we shouldn't be checking anything in the action content because the spec says it's supposed to be a free formed JSON object. If users can put anything they want into the object, then there is nothing much we can check :)

Reply:
Done!